pkg/api/v1/deployment.go

```diff
@@ -84,6 +86,11 @@ func (g *DeploymentGroup) RetrieveDeployment(ctx echo.Context) error {
 
 	deploymentId := ctx.Param("deploymentId")
 	if deployment, err := g.backendRepo.GetDeploymentByExternalId(ctx.Request().Context(), workspace.Id, deploymentId); err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
This doesn't check for bad input (id not being a uuid) and will surface as a 500 when it should probably be a 400.

Check this out.
https://github.com/beam-cloud/beta9/blob/main/pkg/repository/backend_postgres.go#L419-L427

Reply:
nice, will add this